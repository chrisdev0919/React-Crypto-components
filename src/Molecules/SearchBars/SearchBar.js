import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Input } from '../../Atoms/Inputs'
import { Icon } from '../../Atoms/Icons'

const BaseSearchBar = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;

  & > :first-child {
    border-top-left-radius: 5px;
    border-bottom-left-radius: 5px;
  }

  & > :last-child {
    border-top-right-radius: 5px;
    border-bottom-right-radius: 5px;
  }
`
const IconContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 60px;
  height: 40px;
  background-color: ${props => props.theme['gold']};
`

const SearchBar = ({ children, borderColor }) => (
  <BaseSearchBar>
    <Input borderColor={borderColor} />
    <IconContainer>
      <Icon name='search-filled' size='20px' selectable />
    </IconContainer>
  </BaseSearchBar>
)

SearchBar.propTypes = {
  borderColor: PropTypes.string
}

SearchBar.defaultProps = {
  borderColor: 'white'
}

export default SearchBar
