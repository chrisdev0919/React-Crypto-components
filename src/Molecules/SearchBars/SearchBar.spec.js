import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import SearchBar from './SearchBar'

describe('SearchBar component', () => {
  it('renders correctly', () => {
    const component = shallow(<SearchBar />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with borderColor', () => {
    const component = shallow(<SearchBar borderColor='gold' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
