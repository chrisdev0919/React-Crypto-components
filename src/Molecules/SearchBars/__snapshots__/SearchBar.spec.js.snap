// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders correctly 1`] = `
<styled.div>
  <Input
    borderColor="white"
    disabled={false}
    type="text"
    width="100%"
  />
  <styled.div>
    <Icon
      color="gray-5"
      name="search-filled"
      selectable={true}
      size="20px"
      weight={400}
    />
  </styled.div>
</styled.div>
`;

exports[`SearchBar component renders correctly with borderColor 1`] = `
<styled.div>
  <Input
    borderColor="gold"
    disabled={false}
    type="text"
    width="100%"
  />
  <styled.div>
    <Icon
      color="gray-5"
      name="search-filled"
      selectable={true}
      size="20px"
      weight={400}
    />
  </styled.div>
</styled.div>
`;
